### PR TITLE
Move all errno handling to libfuse

### PR DESCRIPTION
**Pull request by akalin-keybase**
This removes a dependency on bazil from various places.

Change reportErr in libfuse to processError, and move the errno
mapping to a filterError function.

### CODE REVIEW

File: kbfsmd/errors_fuse.go [hunk 1]
```diff
@@ -14,20 +14,6 @@ import (
...
-	return fuse.Errno(syscall.EACCES)
-}
-
-var _ fuse.ErrorNumber = ServerErrorWriteAccess{}
-
-// Errno implements the fuse.ErrorNumber interface for ServerErrorWriteAccess.
-func (e ServerErrorWriteAccess) Errno() fuse.Errno {
-	return fuse.Errno(syscall.EACCES)
-}
-
```
Comment (akalin-keybase) [inline kbfsmd/errors_fuse.go]: I'll deal with `MetadataIsFinalError` separately...

**Comment by strib** [comment]
There are a few places in libfs/libdokan, maybe elsewhere, that check for one of the specific error types you are converting.  I think you need to fix those places up to pass the tests.

File: libfuse/folderlist.go [hunk 2]
```diff
@@ -138,11 +140,11 @@ func (fl *FolderList) PathType() libkbfs.PathType {
 func (fl *FolderList) Create(ctx context.Context, req *fuse.CreateRequest, resp *fuse.CreateResponse) (_ fs.Node, _ fs.Handle, err error) {
 	fl.fs.log.CDebugf(ctx, "FL Create")
 	tlfName := tlf.CanonicalName(req.Name)
-	defer func() { fl.reportErr(ctx, libkbfs.WriteMode, tlfName, err) }()
+	defer func() { err = fl.processError(ctx, libkbfs.WriteMode, tlfName, err) }()
 	if strings.HasPrefix(req.Name, "._") {
 		// Quietly ignore writes to special macOS files, without
 		// triggering a notification.
-		return nil, nil, libkbfs.WriteUnsupportedError{}.Errno()
+		return nil, nil, syscall.ENOENT
```
Comment (akalin-keybase) [inline libfuse/folderlist.go]: returning ENOENT suffices here, I think

File: libfuse/fs.go [hunk 3]
```diff
@@ -479,19 +481,19 @@ var _ fs.NodeCreater = (*Root)(nil)
 // Create implements the fs.NodeCreater interface for Root.
 func (r *Root) Create(ctx context.Context, req *fuse.CreateRequest, resp *fuse.CreateResponse) (_ fs.Node, _ fs.Handle, err error) {
 	r.log().CDebugf(ctx, "FS Create")
-	defer func() { r.private.fs.reportErr(ctx, libkbfs.WriteMode, err) }()
+	defer func() { err = r.private.fs.processError(ctx, libkbfs.WriteMode, err) }()
 	if strings.HasPrefix(req.Name, "._") {
 		// Quietly ignore writes to special macOS files, without
 		// triggering a notification.
-		return nil, nil, libkbfs.WriteUnsupportedError{}.Errno()
+		return nil, nil, syscall.ENOENT
```
Comment (akalin-keybase) [inline libfuse/fs.go]: ...and here

**Comment by akalin-keybase** [comment]
I changed the approach, which removes the bazil dep from libkbfs, too. PTAL!

File: libfuse/errors.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+var _ fuse.ErrorNumber = errorWithErrno{}
+
+func (e errorWithErrno) Errno() fuse.Errno {
+	return fuse.Errno(e.errno)
+}
+
+func filterError(err error) error {
+	switch err.(type) {
```
Comment (akalin-keybase) [inline libfuse/errors.go]: I should actually call `errors.Cause` (or whatever it's called) here

Comment (strib) [reply libfuse/errors.go]: Yes please!

**Pull request merged**